File: app/src/main/java/su/linka/pictures/components/ConfirmDialog.java
package su.linka.pictures.components;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;

import androidx.annotation.StringRes;

import su.linka.pictures.Callback;
import su.linka.pictures.R;

public class ConfirmDialog {
    public static void showConfirmDialog(Context context, @StringRes int title, Callback callback){
        AlertDialog.Builder builder = new AlertDialog.Builder(context);
        builder
                .setTitle(title)
                .setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        callback.onFail(null);
                    }
                })
                .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        callback.onDone(null);
                    }
                })
                .show();
    }
}

File: app/src/main/java/su/linka/pictures/components/ConfirmDialog.kt
package su.linka.pictures.components

import android.app.AlertDialog
import android.content.Context
import androidx.annotation.StringRes
import su.linka.pictures.Callback
import su.linka.pictures.R

object ConfirmDialog {

    fun showConfirmDialog(context: Context, @StringRes title: Int, callback: Callback<Any?>) {
        AlertDialog.Builder(context)
            .setTitle(title)
            .setNegativeButton(R.string.cancel) { _, _ ->
                callback.onFail(null)
            }
            .setPositiveButton(R.string.ok) { _, _ ->
                callback.onDone(null)
            }
            .show()
    }
}
